framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -351,6 +353,25 @@ public long getFromValueFromIntRange(AnnotatedTypeMirror atm) {
             case CHAR:
                 from = Character.MIN_VALUE;
                 break;
+            case DECLARED:
+                String qualifiedName = TypesUtils.getQualifiedName((DeclaredType) type).toString();
+                switch (qualifiedName) {
+                    case "java.lang.Integer":
+                        from = Integer.MIN_VALUE;
+                        break;
+                    case "java.lang.Short":
+                        from = Short.MIN_VALUE;
+                        break;
+                    case "java.lang.Byte":
+                        from = Byte.MIN_VALUE;
+                        break;
+                    case "java.lang.Character":
+                        from = Character.MIN_VALUE;
+                        break;
+                    default:
```

Comment:
Use of `default` was a contributor to the problem.  Please make the `default` case throw an exception, and have an explicit test for `"java.lang.Long"`.

Reply:
I can add an explicit case for `Long`. However, I'm not sure that the default case should throw an exception. Do we intend for `@IntRange` only to be applicable to `int`, `Integer`, `long`, etc.? I don't think so - for example, as a user I would expect to be able to write e.g `@IntRange(from=0, to=10) BigInteger x`.

Reply:
What values do you propose as the default `to` and `from` for BigInteger? 

I think it's better for the checker to fail in unhandled cases rather than defaulting to a value that might not make sense.

Reply:
I think it would be better to keep the current behavior, which is to default to `Long.MIN_VALUE` and `Long.MAX_VALUE` for `from` and `to`, respectively. In particular, I am concerned about breaking code using this checker that currently typechecks and is not incorrect.

As written, this change can only make the checker more precise. Adding an exception in the default case, as you suggest, would break any clients that use a single-ended `@IntRange` annotation on a class not on our whitelist (that is, the built in integral types). I don't see a compelling reason to break such clients, nor a compelling reason why they shouldn't exist.

If the Value Checker currently required that `@IntRange` only be written on built-in integral types, then I would support the change you're requesting. However, I don't think it would be wise to permit such annotations in general, but not in this particular use-case.

Reply:
Please make the change.

As I noted, this is the reason for the bug.  It is wise, when you learn of a bug, to fix the root cause that might lead to other problems in the future.

You speculate about existing clients.  If any existing clients exist, the default behavior is as likely to be incorrect as to be correct.  It's better for clients to be notified of a problem than to silently preserve behavior that may be incorrect.